server/cp/macros/update_translation_metadata_macro.py

```diff
@@ -11,19 +11,56 @@
 import logging
 from flask_babel import lazy_gettext
 from superdesk import get_resource_service
+from eve.utils import ParsedRequest
+from apps.archive.common import format_dateline_to_locmmmddsrc
+
 
 logger = logging.getLogger(__name__)
 
 destination_qcodes = ["sfstf", "apfra", "cpstf", "ap---"]
 
 
+def set_dateline(item, dateline):
+    located = item.get("dateline", {}).get("located")
+
+    if located:
+        if located.get("place") and dateline:
+            located["place"] = dateline
+
+        if located.get("city") and dateline.get("name"):
+            located["city"] = dateline["name"]
+
+        if located.get("city_code") and dateline.get("name"):
+            located["city_code"] = dateline["name"]
```

Comment:
seems wrong to use name as city code

Reply:
but that shouldn't change with language anyhow

Reply:
I changed it just because of one reason:
we use [this](https://github.com/superdesk/superdesk-core/blob/develop/apps/archive/common.py#L209) method to create the `dateline['text']` filed and this needs the city code, so if the city_code is not updated then, it will display the same name as before.

Reply:
I see, but you still shouldn't set `city_code=name` right?

Reply:
No doubt, you are right here, seems like we need to update the core functionality?

Reply:
We have used the name attribute as the `city_code`: [here]( https://github.com/superdesk/superdesk-client-core/blob/develop/scripts/apps/authoring/metadata/PlacesService.ts#L77) because we don't have the city code in the Geonames API.